### DISCUSSION TITLE
add a listener for all midi messages

### DISCUSSION THREAD

**Discussion by elmcapp** [djipco/webmidi#265] [Q&A]
I would like to listen to all midi message. currently I have to add a listener for each message type. note-on, note-off, control-change. 

I am using the following and its not working
```
WebMidi.inputs.forEach(input => {
  input.addListener("midimessage", e => console.log(e));
});
```
I also tried the following
```
const myInput = WebMidi.getInputByName("IAC Driver Bus 1");
    myInput.addListener("midimessage", (e) => {
      // console.log(e.note.identifier);
      io.emit("midi_msg", e);
    });
```

**Comment by djipco** [comment]
Your first code excerpt should work fine. This is what I just tried and it works for me in Node:

```js
  WebMidi.inputs.forEach(input => {
    input.addListener("midimessage", e => console.log(e.message.type));
  });
```

Is there anything special with your environment ?
